From b8282c5218c7783f69fe4e945808ee2a0f638722 Mon Sep 17 00:00:00 2001
From: Christophe Priouzeau <christophe.priouzeau@foss.st.com>
Date: Fri, 1 Sep 2023 11:23:04 +0200
Subject: [PATCH 4/4] gcc: Processing_is_tee_symm -Werror=enum-int-mismatch

ta/pkcs11/src/processing_symm.c:28:6: error: conflicting types for 'processing_is_tee_symm' due to enum/integer mismatch; have '_Bool(enum pkcs11_mechanism_id)' [-Werror=enum-int-mismatch]
   28 | bool processing_is_tee_symm(enum pkcs11_mechanism_id proc_id)
      |      ^~~~~~~~~~~~~~~~~~~~~~
In file included from ta/pkcs11/src/processing_symm.c:20:
ta/pkcs11/src/processing.h:123:6: note: previous declaration of 'processing_is_tee_symm' with type '_Bool(uint32_t)' {aka '_Bool(unsigned int)'}
  123 | bool processing_is_tee_symm(uint32_t proc_id);
      |      ^~~~~~~~~~~~~~~~~~~~~~

Signed-off-by: Christophe Priouzeau <christophe.priouzeau@foss.st.com>
---
 ta/pkcs11/src/processing.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ta/pkcs11/src/processing.h b/ta/pkcs11/src/processing.h
index 943514f79..59065535b 100644
--- a/ta/pkcs11/src/processing.h
+++ b/ta/pkcs11/src/processing.h
@@ -120,7 +120,7 @@ enum pkcs11_rc step_asymm_operation(struct pkcs11_session *session,
 /*
  * Symmetric crypto algorithm specific functions
  */
-bool processing_is_tee_symm(uint32_t proc_id);
+bool processing_is_tee_symm(enum pkcs11_mechanism_id proc_id);
 
 enum pkcs11_rc init_symm_operation(struct pkcs11_session *session,
 				   enum processing_func function,
-- 
2.34.1

